workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationService.test.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import { AuthenticatedUser } from '../authenticatedUser';
+import { CreateGroupResponse } from './dynamicAuthorizationInputs/createGroup';
+import { DynamicAuthorizationService } from './dynamicAuthorizationService';
+import { GroupManagementPlugin } from './groupManagementPlugin';
+
+describe('WBCGroupManagemntPlugin', () => {
+  const mockUser: AuthenticatedUser = {
+    id: 'sampleId',
+    roles: []
+  };
+
+  const mockGroupManagementPlugin: GroupManagementPlugin = {
+    createGroup: jest.fn(),
+    deleteGroup: jest.fn(),
+    getUserGroups: jest.fn(),
+    getGroupUsers: jest.fn(),
+    addUserToGroup: jest.fn(),
+    isUserAssignedToGroup: jest.fn(),
+    removeUserFromGroup: jest.fn(),
+    getGroupStatus: jest.fn(),
+    setGroupStatus: jest.fn()
+  };
+
+  let dynamicAuthzService: DynamicAuthorizationService;
+
+  beforeEach(() => {
+    dynamicAuthzService = new DynamicAuthorizationService({
+      groupManagementPlugin: mockGroupManagementPlugin
+    });
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('createGroup', () => {
+    it('returns `created` as true when the group was successfully created', async () => {
+      const groupId = 'groupId';
+      const status = 'active';
+      mockGroupManagementPlugin.createGroup = jest.fn().mockResolvedValue({ data: { groupId } });
```

Comment:
I know the tests are structured this way now, but this would be great in a `beforeEach` method to help cut down on repeated code and allow for overrides.

Reply:
fixed!